### PR TITLE
102/standardize frontend tags

### LINKED ISSUE #102: Standardize frontend tags
Currently we have 3 competing implementations of Club and Event tags, we should collapse this into one common implementation that can serve all 3 uses (such as colours)

### PR DESCRIPTION
**Pull request by wang-em**
## Screenshots
From common component change both event and club info pages now have colour coded tag bubbles.
Before:
club info before

After:
clubs Info

Other pages remain the same: 
Clubs Page

## Purpose
Closes #102 
Removed TAGS map since it is redundant. Colours of the bubbles will be kept with style.

### CODE REVIEW

File: client/components/club-details/EventsHostedByClub.tsx [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import styled from 'styled-components'
 import moment from 'moment-timezone'
-import { Category } from '../Category'
 import { colours, desktopFontSize, device, fontWeight, largerThan, mobile } from '../../styles'
```
Comment (xujustinj) [inline client/components/club-details/EventsHostedByClub.tsx:3]:
```suggestion
import { TagBubble } from '../common/TagBubble'
import { colours, desktopFontSize, device, fontWeight, largerThan, mobile } from '../../styles'
```

this is causing the Vercel deployment to fail

**Review by xujustinj — COMMENTED** [review]
The screenshots look great! I'll take a closer look when the Vercel preview is fixed.

File: client/components/clubs/ClubCard.tsx [hunk 2]
```diff
@@ -75,23 +76,14 @@ const ClubCardDescription = styled.p`
...
     return (
-      <RightSpaceWrapper key={text}>
-        <TagBubble colour={colour}>{text}</TagBubble>
-      </RightSpaceWrapper>
+        <TagBubble colour={colours.tagColours[tag]} key={`club-card-tag-${tag}`}>{tag}</TagBubble>
     )
   } else {
     return (
-      <RightSpaceWrapper key={tag}>
         <TagBubble>{tag}</TagBubble>
```
[the remark this answers is not in the document]
Comment (jeffzh4ng) [reply client/components/clubs/ClubCard.tsx:86]: Is this necessary? Are we rendering `<TagBubble>`s via array?

**Review by jeffzh4ng — APPROVED** [review]
Super clean! LGTM :)

File: client/components/clubs/ClubCard.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (xujustinj) [reply client/components/clubs/ClubCard.tsx:86]: Yep, it's mapped in the tags list on the club card. The other branch of this function already produces a `TagBubble` with a key.

File: client/components/common/TagBubble.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  border-width: ${({ borderWidth }) => borderWidth || 'auto'};
+  box-sizing: border-box;
+  display: flex;
+  flex-grow: 0;
+  flex-shrink: 0;
+  height: ${TAG_HEIGHT};
+  justify-content: center;
+  line-height: ${TAG_HEIGHT};
+  margin-right: 16px;
+  margin-bottom: 1px; // make space between rows of bubbles
```
[the remark this answers is not in the document]
Comment (wang-em) [reply client/components/common/TagBubble.tsx:30]: can this issue be reproduced? I removed the margin bottom and don't see a problem with wrapping

[the remark this answers is not in the document]
Comment (xujustinj) [reply client/components/common/TagBubble.tsx]: The artifacts were weird lines appearing due to the second row of tags bleeding into the first. (We only display the first row). If we cannot reproduce it, the 1px margin can be removed.

[the remark this answers is not in the document]
Comment (wang-em) [reply client/components/common/TagBubble.tsx]: Style wise it's fine without the margin. Just wondering why we only display first row of tags on mobile?

[the remark this answers is not in the document]
Comment (xujustinj) [reply client/components/common/TagBubble.tsx]: It's for the header:

image

We only want a single row of tags, but the width of that row is responsive. To solve this, I wrap any excess tags into a second row and hide the row. There might be a better way to do this that I don't know of

The wrapping should happen on both mobile and desktop

[the remark this answers is not in the document]
Comment (wang-em) [reply client/components/common/TagBubble.tsx]: mobile
Yeah I'm just confused about why only 4 tags show up on mobile. That makes the "excess tags" unusable and doesn't make much sense functionality wise?

[the remark this answers is not in the document]
Comment (xujustinj) [reply client/components/common/TagBubble.tsx]: It matches design.
image

Clicking `+ More` will do something once the filtering changes are in, so the excess tags will eventually be accessible

[the remark this answers is not in the document]
Comment (wang-em) [reply client/components/common/TagBubble.tsx]: Ah that makes sense! the `margin-bottom` has been removed since it didn't change anything

**Review by xujustinj — APPROVED** [review]

**Review by jeffzh4ng — APPROVED** [review]

**Pull request merged**